redical_core/src/queries/event_instance_query.rs

```diff
@@ -438,7 +491,574 @@ mod test {
     use crate::{GeoPoint, KeyValuePair};
     use pretty_assertions_sorted::{assert_eq, assert_eq_sorted};
 
-    use std::collections::HashSet;
+    use std::collections::{HashSet, HashMap};
+
+    #[test]
+    fn test_uid_index_retrieval() {
+        let mut calendar = Calendar::new(String::from("CALENDAR_UID"));
+
+        let event_one = Event::parse_ical("EVENT_ONE", "").unwrap();
+        let event_two = Event::parse_ical("EVENT_TWO", "").unwrap();
+        let event_three = Event::parse_ical("EVENT_THREE", "").unwrap();
+
+        calendar.insert_event(event_one);
+        calendar.insert_event(event_two);
+        calendar.insert_event(event_three);
+        calendar.rebuild_indexes().unwrap();
+
+        let accessor = EventInstanceQueryIndexAccessor::new(&calendar);
+
+        // Positive matching: term exists
+        assert_eq!(
+            accessor.search_uid_index("EVENT_ONE"),
+            InvertedCalendarIndexTerm {
+                events: HashMap::from([
+                    (String::from("EVENT_ONE"), IndexedConclusion::Include(None)),
+                ]),
+            }
+        );
+
+        // TODO: Shouldn't this return an empty event set?
+        // Positive matching: term does not exist
+        assert_eq!(
+            accessor.search_uid_index("EVENT_FOUR"),
+            InvertedCalendarIndexTerm {
+                events: HashMap::from([
+                    (String::from("EVENT_FOUR"), IndexedConclusion::Include(None)),
+                ]),
+            }
+        );
```

Comment:
Yeah very very very good spot 👀🦅, it should return an empty set. 🤔 I think it was covered up because the query pipeline ends up plucking all events from the `Calendar.events` `HashMap` which will naturally filter out those UIDs that do not actually exist. Does not hurt though to tweak `search_uid_index` to only include it if it actually exists though 🤔